### PR TITLE
Es6

### PR DESCRIPTION
**Pull request by jeiholzer**
Update to ES6

### CODE REVIEW

File: advanced-skin/components/register-service.js [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
  *
- * Copyright (c) 2015 Digital Bazaar, Inc. All rights reserved.
+ * Copyright (c) 2015-2017 Digital Bazaar, Inc. All rights reserved.
  */
-define([], function() {
-
-'use strict';
-
-/* @ngInject */
 function factory() {
```
Comment (mattcollier) [inline advanced-skin/components/register-service.js]: combine what's on L16 here with
```
export default function factory () {
```

File: express-route-override/components/main.js [hunk 2]
```diff
@@ -1,15 +1,16 @@
 /*!
  * Copyright (c) 2017 Digital Bazaar, Inc. All rights reserved.
  */
-define([
-  'angular',
-  './test-route-component'
-], function(angular) {
-
-'use strict';
+import angular from 'angular';
```
Comment (mattcollier) [inline express-route-override/components/main.js]: The new convention for this file is to call it `index.js`.  And in the package.json file, for modules like these that are both nodejs and front-end, the `main` property points to the nodesjs index.js (typically './lib/index.js') and the `browser` property points to the `index.js` in `components`.  Typically:
```
{
  main: 'lib',
  browser: 'index.js'
}
```

File: express-route-override/config.js [hunk 3]
```diff
@@ -26,7 +26,9 @@ config.views.vars.siteTitle = config.brand.name;
...
 // pseudo bower package for example
 const rootPath = path.resolve(path.join(__dirname));
-config.requirejs.bower.packages.push({
+config.views.system.packages.push({
   path: path.join(rootPath, 'components'),
-  manifest: path.join(rootPath, 'bower.json')
+  manifest: path.join(rootPath, 'package.json')
 });
+
+config.views.system.importAllIgnore.push('bootstrap');
```
Comment (mattcollier) [inline express-route-override/config.js]: we shouldn't need this anywhere anymore as this was a work around.  If boostrap is a requirement you'll want to use: https://github.com/digitalbazaar/bedrock-angular-bootstrap and not have a direct bootstrap dependency anywher.

File: angular-basic/lib/config.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ var path = require('path');
 
 // angular-basic pseudo bower package
```
Comment (mattcollier) [inline angular-basic/lib/config.js]: `pseudo bower package` is no longer a thing.  Change this to `pseudo package` everywhere.

**Review by dlongley — CHANGES_REQUESTED** [review]
I think this PR looks good, but I would recommend one more change that's important for people who copy these examples: Every use of `var` should be replaced with `const` (or `let` if and only if the variable is not initialized immediately or is reassigned).

**Comment by dlongley** [comment]
We'll want to squash all of those `upgrade files` commits as well -- when we're ready to merge. We can't do a complete squash to a single commit because there's multiple authorship here.

**Comment by mattcollier** [comment]
@davidlehn @msporny @jeiholzer @dlongley on the topic of the use of `const`.  I recently updated the eslint rules to help us with this.  Eslint will return an error if `const` has been used incorrectly (variable has been reassigned).

https://github.com/digitalbazaar/bedrock-dev/commit/8a421b612a3f337ba14df87535a5515f2bdb8db1

Eslint will also recommend that `const` be used when possible.  There are some additional settings related to this behavior surrounding destructuring which we can look into later.

http://eslint.org/docs/rules/prefer-const

**Comment by dlongley** [comment]
The skinned component in the advanced-skin example was lost in the latest commit that was about changing `var` to `const`. Was that intentional?

**Comment by mattcollier** [comment]
@jeiholzer Please review convention wrt commit messages: https://github.com/digitalbazaar/bedrock/blob/master/CONTRIBUTING.md#commit-messages

File: advanced-skin/components/password-confirmation-directive.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+/*!
+ * Password confirmation directive.
+ *
+ * Copyright (c) 2015-2017 Digital Bazaar, Inc. All rights reserved.
+ */
+export default function factory($timeout) {
```
Comment (mattcollier) [inline advanced-skin/components/password-confirmation-directive.js]: @jeiholzer this directive can eliminated here.  We have added a directive in bedrock-angular-form that does the same thing.   You can see how it's implemented here:

https://github.com/digitalbazaar/bedrock-angular-authn-password/blob/material/password-reset-component.html#L34

https://github.com/digitalbazaar/bedrock-angular-form/blob/master/validator-same-as-directive.js

**Review by dlongley — APPROVED** [review]
LGTM other than fixing up commit messages and doing some squashing (without losing authorship). @mattcollier  can you work with @jeiholzer on that (the squashing)?

**Comment by jeiholzer** [comment]
@dlongley @mattcollier @dlehn all comments have been addressed; ready for another review

**Comment by mattcollier** [comment]
@jeiholzer some comments have not been addressed, one in `express-route-override/config.js` and one in `advanced-skin/components/register-service.js` also, other that you have `show outdated` to see.

You should be looking at the `conversation` tab here: https://github.com/digitalbazaar/bedrock-examples/pull/3

**Comment by jeiholzer** [comment]
I think everything has been addressed but I did leave config.views.system.importAllIgnore.push('bootstrap'); in basic-modals-example because even when switching to using #material branch it got to a point where the example worked but not with modals.It seemed better to not rely on branches that will eventually go away. Will update when '#material' work is done. Let me know if you disagree.

**Comment by jeiholzer** [comment]
Ok, I think I have addressed everything and modified the appropriate examples I was working on to Angular Material

**Comment by dlongley** [comment]
@jeiholzer,

This PR needs conflict resolution then I believe it's ready to be merged in.

cc: @mattcollier

**Pull request merged**